### PR TITLE
Address Format Docs

### PR DESCRIPTION
**Pull request by ingalls**
### Context

Doc specific to how we expect address data to be formatted as well as pitfalls that a user may encounter. This document will be a sister document to the existing data-sources.md document, just with a specific eye towards addresses.

### Summary of Changes
- [x] Add addresses.md formatting document

cc @mapbox/search

### CODE REVIEW

File: docs/addresses.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+```
+
+Note: Interpolation properties all contain `l` (left) and `r` (right) prefixes. These
+left/right prefixes are based on the direction of the linestring from the 0th coordinate
+onward. When visualizing these values, ensure you check the orientation of the linestring
+as a common gotcha is using the L/R of your orientation to the feature on your screen.
+
+- `properties.carmen:rangetype`
+    - REQUIRED: Carmen has the potential to support multiple forms of interpolation,
+      however at the moment only `tiger` is currently implemented and supported.
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: What do you propose? To me this reads as `tiger` being explicitly the only option

File: docs/addresses.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    - As per above, the `geometry.coordinates` array must be parallel to, and equal in length with
+      the `properties.carmen:intersections` array
+    - CAN be a `GeometryCollection` but if so it must follow the rules of defined in the
+    [Combined Features](#combined-features) section of the document. `GeometryCollections`
+    cannot follow the format as in the example above.
+
+
+### Combined Features
+
+Combined features allow a feature to be indexed in a more optimized form. By default, PT2ITP
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: Hmm address clusters only refer to the numbers themselves. I've reworded a little bit.

File: docs/addresses.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+      are all valid examples of supported addresses
+    - The `properties.carmen:addressnumber` is a parallel array to the coordinates array
+      found at `geometry.coordinates`. This means that the lengths must be the same and that
+      the address number of a given element within this array shares the coordinate with the
+      element in the parallel geometry array. For example the number at `properties.carmen:addressnumber[0]`
+      has its corresponding coordinate at `geometry.coordinates[0]`.
+- `geometry.type`
+    - REQUIRED: A geojson `MultiPoint` type
+    - As per above, the `geometry.coordinates` array must be parallel two, and equal in length with
+      the `properties.addressnumber` array.
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: I think we should be explicit when describing the contents of each, especially for folks just skimming the doc, like the GeoJSON spec doc does.

File: docs/addresses.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+   - REQUIRED: same as `parityl` except for the right side
+- `properties.carmen:rfromhn`
+   - REQUIRED: same as `lfromhn` except for the right side
+- `properties.carmen:rtohn`
+   - REQUIRED: same as `ltohn` except for the right side
+- `geometry.type`
+    - REQUIRED: A geojson `MultiLineString` type
+    - As per above, the `geometry.coordinates` array must be parallel to, and equal in length with
+      the `parity` & `from/to` arrays.
+    - CAN be a `GeometryCollection` but if so it must follow the rules of defined in the
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: Per above, I think we should be explicit here, changed the order per your suggestion

File: docs/addresses.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+```
+
+- `properties.carmen:intersections`
+    - REQUIRED: A flat array containing a single name of each intersecting street
+      If the intersecting street has multiple synonyms that you want the user to be
+      able to search for, each synonym should have it's own element in the array.
+- `geometry.type`
+    - REQUIRED: A geojson `MultiPoint` type
+    - As per above, the `geometry.coordinates` array must be parallel to, and equal in length with
+      the `properties.carmen:intersections` array
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: Yes

File: docs/addresses.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+address MultiPoint geometry in position 1, then the properties would look like:
+
+```JSON
+{
+    "carmen:addressnumber": [null, [100, 101, 102]]
+}
+```
+
+
+```JSON
```
[the remark this answers is not in the document]
Comment (ingalls) [reply docs/addresses.md]: @lizziegooding I don't think we should document or trust the output order from pt2itp. There is no guarantee of output order, and anything apart from one-off scripts should not be built to rely on this behavior. Neither carmen, nor pt2itp itself do, instead doing what you describe, finding the element number dynamically.

File: docs/addresses.md [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+      found at `geometry.coordinates`. This means that the lengths must be the same and that
+      the address number of a given element within this array will have the same
+      element position as the coordinates in the parallel geometry array. For example the number at `properties.carmen:addressnumber[0]`
+      has its corresponding coordinates at `geometry.coordinates[0]`.
+- `geometry.type`
+    - REQUIRED: A geojson `MultiPoint` type
+    - CAN be a `GeometryCollection` but if so it must follow the rules of defined in the
+    [Combined Features](#combined-features) section of the document. `GeometryCollections`
+    cannot follow the format as in the example above.
+    - As per above, the `geometry.coordinates` array must be parallel two, and equal in length with
```
Comment (lizziegooding) [inline docs/addresses.md]: s/two/to

File: docs/addresses.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lizziegooding) [reply docs/addresses.md]: > - REQUIRED: The interpolation format. Accepted values: `tiger`

File: docs/addresses.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (lizziegooding) [reply docs/addresses.md]: I personally find the following more clear, especially given that internally at least we output everything as a `GeometryCollection`:

> - `geometry.type`
    - REQUIRED: A geojson `MultiLineString` or `GeometryCollection` type. If it's a `GeometryCollection` type,  it must follow the rules of defined in the...

You could also explicitly title the Multi<Feat> examples in each section and the GeometryCollection type to make it easier to navigate

File: docs/addresses.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lizziegooding) [reply docs/addresses.md]: The first part of my comment still stands, think we should address: 
> an we add some text to explain that this is an example of a combined address cluster feature with all three of the address types mentioned above? The example above is very helpful, but the transition is abrupt. 

Re the address order, can we note that the order is not guaranteed and therefore any scripts should find the relevant sub geometry in X manner and what we do most often internally if you feel we're consistent?

File: docs/addresses.md [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+            "carmen:center": [ -76.9989961, 38.8235612 ],
+    },
+    "geometry": {
+        "type": "MultiPoint",
+        "coordinates": [
+            [ -76.9994694, 38.8240466 ],
+            [ -76.9994929, 38.8237324 ],
+            [ -76.9994077, 38.82405 ],
+            [ -76.9994363, 38.8237197 ],
+            [ -76.9993205, 38.8240264 ],
```
Comment (mattciferri) [inline docs/addresses.md]: @ingalls Went through this for my own learning and found a few examples of trailing commas that prevent these features from cleanly loading into geojson.io. There's three in this one.

File: docs/addresses.md [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+        "carmen:rangetype": "tiger",
+        "carmen:parityl": [ "E" ],
+        "carmen:lfromhn": [ 600 ],
+        "carmen:ltohn": [ 640 ],
+        "carmen:parityr": [ "O" ],
+        "carmen:rfromhn": [ 601 ],
+        "carmen:rtohn": [ 631 ],
+        "carmen:text": "Galveston Street Southeast,US Highway 123",
+        "carmen:geocoder_stack": "us",
+        "carmen:center": [ -76.9989961, 38.8235612 ],
```
Comment (mattciferri) [inline docs/addresses.md]: Here

File: docs/addresses.md [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+            ]
+        ],
+        "carmen:addressnumber": [
+            null,
+            [
+                "600",
+                "601",
+                "602",
+                "603",
+                "604",
```
Comment (mattciferri) [inline docs/addresses.md]: Here

File: docs/addresses.md [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+        "carmen:rtohn": [
+            [
+                631
+            ],
+            null,
+            null
+        ],
+        "carmen:text": "Galveston Street Southeast",
+        "carmen:geocoder_stack": "us",
+        "carmen:center": [ -76.9989961, 38.8235612 ],
```
Comment (mattciferri) [inline docs/addresses.md]: Here

File: docs/addresses.md [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+                [ -76.9985091, 38.8234421 ]
+            ]]
+        },{
+            "type": "MultiPoint",
+            "coordinates": [
+                [ -76.9994694, 38.8240466 ],
+                [ -76.9994929, 38.8237324 ],
+                [ -76.9994077, 38.82405 ],
+                [ -76.9994363, 38.8237197 ],
+                [ -76.9993205, 38.8240264 ],
```
Comment (mattciferri) [inline docs/addresses.md]: Last one

**Pull request merged**